File: backend/src/models/Job.ts
import mongoose, { Schema, InferSchemaType } from 'mongoose';

const JobSchema = new Schema({
  sourceUrl: { type: String, required: true },
  status: { type: String, enum: ['pending', 'processing', 'done', 'error'], default: 'pending' },
  error: { type: String },
  createdAt: { type: Date, default: Date.now },
});

export type JobDoc = InferSchemaType<typeof JobSchema> & { _id: string };

export default mongoose.model('Job', JobSchema);

File: backend_py/app/models/job.py
from datetime import datetime, timezone
from typing import Literal, Optional

from beanie import Document
from pydantic import Field

JobStatus = Literal["pending", "processing", "done", "error"]


def _utcnow() -> datetime:
    return datetime.now(timezone.utc)


class Job(Document):
    """Mirrors the Mongoose `Job` model.

    NOTE: field names are camelCase on purpose. The documents were written by
    the previous Node/Mongoose backend and the frontend reads the same keys, so
    keeping Python attribute == Mongo field == JSON key means zero data
    migration and zero frontend changes.
    """

    sourceUrl: str
    status: JobStatus = "pending"
    error: Optional[str] = None
    createdAt: datetime = Field(default_factory=_utcnow)

    class Settings:
        name = "jobs"
